scripts/backup-upload-aws.js

```diff
@@ -0,0 +1,122 @@
+const fs = require('fs');
+const AWSService = require('../modules/service/aws-service');
+const { execSync } = require('child_process');
+const path = require('path');
+const argv = require('minimist')(process.argv.slice(2));
+const { lstatSync, readdirSync } = require('fs');
+const { join } = require('path');
+
+
+function walkSync(currentDirPath, callback) {
+    fs.readdirSync(currentDirPath).forEach((name) => {
+        const filePath = path.join(currentDirPath, name);
+        const stat = fs.statSync(filePath);
+        if (stat.isFile()) {
+            callback(filePath, stat);
+        } else if (stat.isDirectory()) {
+            walkSync(filePath, callback);
+        }
+    });
+}
+
+function isEmpty(path) {
+    return fs.readdirSync(path).length === 0;
+}
+
+if (!argv.config) {
+    argv.config = '.origintrail_noderc';
+}
+
+if (!argv.configDir) {
+    argv.configDir = '../data/';
+}
+
+if (!argv.backupDirectory) {
+    argv.backupDirectory = '../backup/';
+}
+
+if (!argv.AWSAccessKeyId) {
+    throw Error('Please provide AWS access key id.');
+}
+
+if (!argv.AWSSecretAccessKey) {
+    throw Error('Please provide AWS secret key.');
+}
+
+if (!argv.AWSBucketName) {
+    throw Error('Please provide AWS bucket name.');
+}
+
+try {
+    if (fs.existsSync(argv.backupDirectory) && !isEmpty(argv.backupDirectory)) {
```

Comment:
So if there already is a backup in the `/ot-node/backup/` folder, we'll delete all of it?

Reply:
Yes, we are sorting directories by name in the backup folder and take the first directory. If a user creates directory in the backup folder we can take a wrong one.